Code/Engine/RendererFoundation/Resources/DynamicBuffer.h

```diff
@@ -0,0 +1,88 @@
+#pragma once
+
+#include <RendererFoundation/Descriptors/Descriptors.h>
+
+class EZ_RENDERERFOUNDATION_DLL ezGALDynamicBuffer
+{
+public:
+  template <typename U>
+  ezUInt32 Allocate(const U& userData, ezUInt32 uiCount)
+  {
+    static_assert(sizeof(U) <= sizeof(ezUInt64), "userData is too large");
+    ezUInt64 uiUserData = 0;
+    *reinterpret_cast<U*>(&uiUserData) = userData;
+
+    return Allocate(uiUserData, uiCount);
+  }
+
+  void Deallocate(ezUInt32 uiOffset);
```

Comment:
Add docs talking about holes / compaction. Also why can I only deallocate at an offset? Shouldn't I also be able to give a range?

Reply:
An allocation is identified by its offset just like you would identify a regular allocation by its pointer.